tests/unit/migration/test_azure_credentials.py

```diff
@@ -0,0 +1,93 @@
+from unittest.mock import MagicMock, create_autospec
+
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.service.catalog import (
+    AwsIamRole,
+    AzureManagedIdentity,
+    AzureServicePrincipal,
+    StorageCredentialInfo,
+)
+
+from databricks.labs.ucx.assessment.azure import StoragePermissionMapping
+from databricks.labs.ucx.migration.azure_credentials import (
+    AzureServicePrincipalMigration,
+)
+
+
+def test_list_storage_credentials():
+    w = create_autospec(WorkspaceClient)
+
+    w.storage_credentials.list.return_value = [
+        StorageCredentialInfo(aws_iam_role=AwsIamRole(role_arn="arn:aws:iam::123456789012:role/example-role-name")),
+        StorageCredentialInfo(
+            azure_managed_identity=AzureManagedIdentity(
+                access_connector_id="/subscriptions/.../providers/Microsoft.Databricks/..."
+            )
+        ),
+        StorageCredentialInfo(
+            azure_service_principal=AzureServicePrincipal(
+                application_id="b6420590-5e1c-4426-8950-a94cbe9b6115",
+                directory_id="62e43d7d-df53-4c64-86ed-c2c1a3ac60c3",
+                client_secret="secret",
+            )
+        ),
+    ]
+
+    sp_migration = AzureServicePrincipalMigration(MagicMock(), w, MagicMock(), MagicMock())
+
+    expected = {"b6420590-5e1c-4426-8950-a94cbe9b6115"}
+    sp_migration._list_storage_credentials()
+
+    assert expected == sp_migration._list_storage_credentials()
+
+
+def test_sp_in_storage_credentials():
```

Comment:
Consider using @pytest.mark.parametrize for multiple runs by the same test.

Reply:
used @pytest.mark.parametrize in other tests. And this test is actually removed since `_check_sp_in_storage_credentials` is replaced by list comprehension